scrapy/core/downloader/handlers/s3.py

```diff
@@ -49,14 +47,32 @@ def __init__(self, settings, aws_access_key_id=None, aws_secret_access_key=None,
         # If no credentials could be found anywhere,
         # consider this an anonymous connection request by default;
         # unless 'anon' was set explicitly (True/False).
-        anon = kw.get('anon', None)
+        anon = kw.get('anon')
         if anon is None and not aws_access_key_id and not aws_secret_access_key:
             kw['anon'] = True
+        self.anon = kw.get('anon')
 
+        self._signer = None
         try:
-            self.conn = _S3Connection(aws_access_key_id, aws_secret_access_key, **kw)
-        except Exception as ex:
-            raise NotConfigured(str(ex))
+            import botocore.auth
+            import botocore.credentials
+        except ImportError:
+            if six.PY3:
+                raise NotConfigured("missing botocore library")
+            _S3Connection = get_s3_connection()
+            if _S3Connection is None:
+                raise NotConfigured("missing botocore or boto library")
+            try:
+                self.conn = _S3Connection(
+                    aws_access_key_id, aws_secret_access_key, **kw)
```

Comment:
Though I don't know if this even matters, because looking [here](https://github.com/scrapy/scrapy/blob/master/scrapy/core/downloader/handlers/__init__.py#L45) the download never gets any other argument apart from the Settings in production code, only in the test code.

Also, looks like the previous code only mutates the keyword arguments for the `anon` case, which from the comments seems to be inferred when no keys are provided.

So, I guess this would only break for users who are subclassing this and upgrading to Python 3, and perhaps that's acceptable (better than not working in PY3 anyway).